node/sc/mainbridge_test.go

```diff
@@ -64,7 +64,6 @@ func testNewMainBridge(t *testing.T) *MainBridge {
 // testBlockChain returns a test BlockChain with initial values
 func testBlockChain(t *testing.T) *blockchain.BlockChain {
 	db := database.NewMemoryDBManager()
-	defer db.Close()
```

Comment:
Is this on purpose?

Reply:
Yes. As you can see `testBlockChain` creates database inside the function and then use it as a member variable of `Blockchain` which it returns to the caller. Therefore, if `Close` is called inside `testBlockChain`, at the time when the caller receives `Blockchain`, database is closed already. It was not a problem before, since `memDB` did nothing when `Close` is called before. But now resource is deallocated when `Close` is called and this revealed underlying problem.

Maybe we can return `db.Close` to close the database gracefully, however, as it is an in-memory only structure, I didn't implement code to close memory database.

Reply:
Thank you! Understood.